src/transformers/models/beit/configuration_beit.py

```diff
@@ -100,6 +99,22 @@ class BeitConfig(PretrainedConfig):
             Whether to concatenate the output of the auxiliary head with the input before the classification layer.
         semantic_loss_ignore_index (`int`, *optional*, defaults to 255):
             The index that is ignored by the loss function of the semantic segmentation model.
+        out_features (`List[str]`, *optional*):
+            If used as backbone, list of features to output. Can be any of `"stem"`, `"stage1"`, `"stage2"`, etc.
+            (depending on how many stages the model has). If unset and `out_indices` is set, will default to the
+            corresponding stages. If unset and `out_indices` is unset, will default to the last stage.
+        out_indices (`List[int]`, *optional*):
+            If used as backbone, list of indices of features to output. Can be any of 0, 1, 2, etc. (depending on how
+            many stages the model has). If unset and `out_features` is set, will default to the corresponding stages.
+            If unset and `out_features` is unset, will default to the last stage.
+
+            Can also indicate the indices of the feature maps to use for semantic segmentation.
```

Comment:
No no no no no. Having flags with conditional meanings is not a good design.

Moreover, the mixin will update `out_indices` if the user tries to set `out_features` but there's no control on whether or not this is being used as a backbone, leading to potentially unexpected behaviour. 

As a start, this is how I would try to handle this: 

* Remove `out_indices` as an input argument to `__init__`
* Add `semantic_out_indices` as a new argument with the default `None`

* If `semantic_out_indices` is set and `out_indices` is passed in as a kwarg, raise an exception
* If `semantic_out_indices` is `None` and `out_indices` is passed in as a kwarg, set `semantic_out_indices` to `out_indices` 
* If `semantic_out_indices` is `None` and `out_indices` isn't in the kwargs, the `semantic_out_indices` defaults to  `[3, 5, 7, 9]`
* Update model config to not save `out_indices` out including a comment explaining why

You'll need to add tests to make sure that the config and model are properly created with fully backwards compatibility with old configs, as well as fully configurable when creating a model `from_pretrained` and kwargs e.g. `BeitBackbone.from_pretrained(checkpoint, out_indices=(-2, -1))` or  `BeitBackbone.from_pretrained(checkpoint, semantic_out_indices=(-2, -1))`.